### ISSUE TITLE
PermitDynamicIf call ambiguous if trigger parameter is a state

### ISSUE THREAD

**Issue by DeepakParamkusam** [dotnet-state-machine/stateless#380]
Compiler complains that PermitDynamicIf calls in below code are ambiguous-
```
static void Main(string[] args)
{
    var fsm = new StateMachine<State, Trigger>(State.Off);
    StateMachine<State, Trigger>.TriggerWithParameters<State> pressTrigger = fsm.SetTriggerParameters<State>(Trigger.Press);
    StateMachine<State, Trigger>.TriggerWithParameters<State> smashTrigger = fsm.SetTriggerParameters<State>(Trigger.Smash);

    fsm.Configure(State.Off)
        .PermitDynamicIf(pressTrigger, state => { return state; })  // CALL AMBIGUOUS
        .PermitDynamicIf(smashTrigger, GetState);                   // CALL AMBIGUOUS

    while (true)
    {
        Console.WriteLine("Switch is in state: " + fsm.State);
        var pressed = Console.ReadKey(true).KeyChar;

        if (pressed != 'a') break;

        fsm.Fire(smashTrigger, State.Destroyed);
    }
}

public static State GetState(State state)
{
    return state == State.Off ? State.Destroyed : State.On;
}

public enum State
{
    On,
    Off,
    Destroyed
}

public enum Trigger
{
    Press,
    Smash
}
```

Compiler error-

`The call is ambiguous between the following methods or properties: 'StateMachine<TState, TTrigger>.StateConfiguration.PermitDynamicIf<TArg0>(StateMachine<TState, TTrigger>.TriggerWithParameters<TArg0>, Func<TArg0, TState>, DynamicStateInfos, params Tuple<Func<bool>, string>[])' and 'StateMachine<TState, TTrigger>.StateConfiguration.PermitDynamicIf<TArg0>(StateMachine<TState, TTrigger>.TriggerWithParameters<TArg0>, Func<TArg0, TState>, DynamicStateInfos, params Tuple<Func<TArg0, bool>, string>[])'`

The ambiguity seems to be in the optional params. Am I doing something wrong? 

Using .NET Core 3.1, Stateless v5.1.2

**Comment by DeepakParamkusam** [comment]
It compiles if the tuple is explicitly cast -
```
fsm.Configure(State.Off)
    .PermitDynamicIf(pressTrigger, state => state, null, (Tuple<Func<bool>,string>)null)
    .PermitDynamicIf(smashTrigger, GetState, null, (Tuple<Func<State,bool>,string>)null);
```
Seems a bit inelegant. Is this the intended way? I'll close the issue if yes.

**Comment by HenningNT** [comment]
It's not the intended way, that looks a bit off...
I guess all those optional params can make it a bit muddy.
I don't know if we'll do something about right away, though.

**Comment by HenningNT** [comment]
Looking at it again, I wonder where the guard function is supposed to go...
I'd better look into it a bit, thanks for the report!

**Comment by scottgorlick** [comment]
please check my PR as a potential fix that meets the overloads you describe.

**Comment by HenningNT** [comment]
@DeepakParamkusam Would you mind checking how the latest code works for you?
I've tried to create a pre-release, but I seems there is a problem uploading the symbols package to nuget, I don't know if 5.1.3 is successfully uploaded and available.

**Comment by DeepakParamkusam** [comment]
I can confirm that the sample code in my first comment successfully compiles and runs using v5.1.3-dev-00513.  
Thanks for the fix @scottgorlick !

**Comment by HenningNT** [comment]
Thank you for testing it, I have released an official 5.1.3 version.
@scottgorlick Thank you for your contribution!

**Comment by scottgorlick** [comment]
very happy to help.  I'm looking at adding Stateless to my solution so i wanted to see how it was all put together... figured i could fix this low hanging fruit and then move onto a more complicated issue... once i get that done, i'll be very comfortable implementing in my solution.
